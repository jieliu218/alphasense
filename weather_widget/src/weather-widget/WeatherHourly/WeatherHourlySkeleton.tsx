import { Box, Divider, Skeleton, Stack } from "@mui/material";

const WeatherHourlySkeleton = ({ textColor }: { textColor: string }) => {
  return (
    <Box>
      <Divider sx={{ mt: 2, mb: 2, color: textColor }}>Hourly</Divider>
      <Stack
        direction="column"
        justifyContent="justify-content"
        alignItems="center"
        spacing={1}
      >
        <Skeleton variant="rectangular" width={340} height={153} />
        <Skeleton variant="rectangular" width={340} height={153} />
        <Skeleton variant="rectangular" width={340} height={153} />
      </Stack>
    </Box>
  );
};

export default WeatherHourlySkeleton;
