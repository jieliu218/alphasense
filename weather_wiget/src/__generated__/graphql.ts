import { TypedDocumentNode as DocumentNode } from "@graphql-typed-document-node/core";
export type Maybe<T> = T | null;
export type InputMaybe<T> = Maybe<T>;
export type Exact<T extends { [key: string]: unknown }> = {
  [K in keyof T]: T[K];
};
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & {
  [SubKey in K]?: Maybe<T[SubKey]>;
};
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & {
  [SubKey in K]: Maybe<T[SubKey]>;
};
export type MakeEmpty<
  T extends { [key: string]: unknown },
  K extends keyof T,
> = { [_ in K]?: never };
export type Incremental<T> =
  | T
  | {
      [P in keyof T]?: P extends " $fragmentName" | "__typename" ? T[P] : never;
    };
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: { input: string; output: string };
  String: { input: string; output: string };
  Boolean: { input: boolean; output: boolean };
  Int: { input: number; output: number };
  Float: { input: number; output: number };
};

export type CurrentUnits = {
  __typename?: "CurrentUnits";
  interval?: Maybe<Scalars["String"]["output"]>;
  relativeHumidity2m?: Maybe<Scalars["String"]["output"]>;
  temperature2m?: Maybe<Scalars["String"]["output"]>;
  time?: Maybe<Scalars["String"]["output"]>;
  windSpeed10m?: Maybe<Scalars["String"]["output"]>;
};

export type CurrentWeather = {
  __typename?: "CurrentWeather";
  interval?: Maybe<Scalars["Int"]["output"]>;
  relativeHumidity2m?: Maybe<Scalars["Int"]["output"]>;
  temperature2m?: Maybe<Scalars["Float"]["output"]>;
  time?: Maybe<Scalars["String"]["output"]>;
  windSpeed10m?: Maybe<Scalars["Float"]["output"]>;
};

export type HourlyUnits = {
  __typename?: "HourlyUnits";
  relativeHumidity2m?: Maybe<Scalars["String"]["output"]>;
  temperature2m?: Maybe<Scalars["String"]["output"]>;
  time?: Maybe<Scalars["String"]["output"]>;
  windSpeed10m?: Maybe<Scalars["String"]["output"]>;
};

export type HourlyWeather = {
  __typename?: "HourlyWeather";
  relativeHumidity2m?: Maybe<Array<Maybe<Scalars["Int"]["output"]>>>;
  temperature2m?: Maybe<Array<Maybe<Scalars["Float"]["output"]>>>;
  time?: Maybe<Array<Maybe<Scalars["String"]["output"]>>>;
  windSpeed10m?: Maybe<Array<Maybe<Scalars["Float"]["output"]>>>;
};

export type Query = {
  __typename?: "Query";
  getWeatherData?: Maybe<WeatherData>;
};

export type QueryGetWeatherDataArgs = {
  latitude: Scalars["Float"]["input"];
  longitude: Scalars["Float"]["input"];
};

export type WeatherData = {
  __typename?: "WeatherData";
  current?: Maybe<CurrentWeather>;
  currentUnits?: Maybe<CurrentUnits>;
  elevation?: Maybe<Scalars["Float"]["output"]>;
  generationtimeMs?: Maybe<Scalars["Float"]["output"]>;
  hourly?: Maybe<HourlyWeather>;
  hourlyUnits?: Maybe<HourlyUnits>;
  latitude?: Maybe<Scalars["Float"]["output"]>;
  longitude?: Maybe<Scalars["Float"]["output"]>;
  timezone?: Maybe<Scalars["String"]["output"]>;
  timezoneAbbreviation?: Maybe<Scalars["String"]["output"]>;
  utcOffsetSeconds?: Maybe<Scalars["Int"]["output"]>;
};

export type WeatherCurrentFFragment = {
  __typename?: "WeatherData";
  current?: {
    __typename?: "CurrentWeather";
    time?: string | null;
    interval?: number | null;
    temperature2m?: number | null;
    windSpeed10m?: number | null;
    relativeHumidity2m?: number | null;
  } | null;
  currentUnits?: {
    __typename?: "CurrentUnits";
    time?: string | null;
    interval?: string | null;
    temperature2m?: string | null;
    windSpeed10m?: string | null;
    relativeHumidity2m?: string | null;
  } | null;
} & { " $fragmentName"?: "WeatherCurrentFFragment" };

export type WeatherHourlyFFragment = {
  __typename?: "WeatherData";
  hourly?: {
    __typename?: "HourlyWeather";
    time?: Array<string | null> | null;
    temperature2m?: Array<number | null> | null;
    relativeHumidity2m?: Array<number | null> | null;
    windSpeed10m?: Array<number | null> | null;
  } | null;
  hourlyUnits?: {
    __typename?: "HourlyUnits";
    time?: string | null;
    temperature2m?: string | null;
    relativeHumidity2m?: string | null;
    windSpeed10m?: string | null;
  } | null;
} & { " $fragmentName"?: "WeatherHourlyFFragment" };

export type GetWeatherDataQueryVariables = Exact<{
  latitude: Scalars["Float"]["input"];
  longitude: Scalars["Float"]["input"];
  showCurrent: Scalars["Boolean"]["input"];
  showHourly: Scalars["Boolean"]["input"];
}>;

export type GetWeatherDataQuery = {
  __typename?: "Query";
  getWeatherData?:
    | ({
        __typename?: "WeatherData";
        latitude?: number | null;
        longitude?: number | null;
      } & {
        " $fragmentRefs"?: {
          WeatherCurrentFFragment: WeatherCurrentFFragment;
          WeatherHourlyFFragment: WeatherHourlyFFragment;
        };
      })
    | null;
};

export const WeatherCurrentFFragmentDoc = {
  kind: "Document",
  definitions: [
    {
      kind: "FragmentDefinition",
      name: { kind: "Name", value: "WeatherCurrentF" },
      typeCondition: {
        kind: "NamedType",
        name: { kind: "Name", value: "WeatherData" },
      },
      selectionSet: {
        kind: "SelectionSet",
        selections: [
          {
            kind: "Field",
            name: { kind: "Name", value: "current" },
            selectionSet: {
              kind: "SelectionSet",
              selections: [
                { kind: "Field", name: { kind: "Name", value: "time" } },
                { kind: "Field", name: { kind: "Name", value: "interval" } },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "temperature2m" },
                },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "windSpeed10m" },
                },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "relativeHumidity2m" },
                },
              ],
            },
          },
          {
            kind: "Field",
            name: { kind: "Name", value: "currentUnits" },
            selectionSet: {
              kind: "SelectionSet",
              selections: [
                { kind: "Field", name: { kind: "Name", value: "time" } },
                { kind: "Field", name: { kind: "Name", value: "interval" } },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "temperature2m" },
                },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "windSpeed10m" },
                },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "relativeHumidity2m" },
                },
              ],
            },
          },
        ],
      },
    },
  ],
} as unknown as DocumentNode<WeatherCurrentFFragment, unknown>;
export const WeatherHourlyFFragmentDoc = {
  kind: "Document",
  definitions: [
    {
      kind: "FragmentDefinition",
      name: { kind: "Name", value: "WeatherHourlyF" },
      typeCondition: {
        kind: "NamedType",
        name: { kind: "Name", value: "WeatherData" },
      },
      selectionSet: {
        kind: "SelectionSet",
        selections: [
          {
            kind: "Field",
            name: { kind: "Name", value: "hourly" },
            selectionSet: {
              kind: "SelectionSet",
              selections: [
                { kind: "Field", name: { kind: "Name", value: "time" } },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "temperature2m" },
                },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "relativeHumidity2m" },
                },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "windSpeed10m" },
                },
              ],
            },
          },
          {
            kind: "Field",
            name: { kind: "Name", value: "hourlyUnits" },
            selectionSet: {
              kind: "SelectionSet",
              selections: [
                { kind: "Field", name: { kind: "Name", value: "time" } },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "temperature2m" },
                },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "relativeHumidity2m" },
                },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "windSpeed10m" },
                },
              ],
            },
          },
        ],
      },
    },
  ],
} as unknown as DocumentNode<WeatherHourlyFFragment, unknown>;
export const GetWeatherDataDocument = {
  kind: "Document",
  definitions: [
    {
      kind: "OperationDefinition",
      operation: "query",
      name: { kind: "Name", value: "GetWeatherData" },
      variableDefinitions: [
        {
          kind: "VariableDefinition",
          variable: {
            kind: "Variable",
            name: { kind: "Name", value: "latitude" },
          },
          type: {
            kind: "NonNullType",
            type: { kind: "NamedType", name: { kind: "Name", value: "Float" } },
          },
        },
        {
          kind: "VariableDefinition",
          variable: {
            kind: "Variable",
            name: { kind: "Name", value: "longitude" },
          },
          type: {
            kind: "NonNullType",
            type: { kind: "NamedType", name: { kind: "Name", value: "Float" } },
          },
        },
        {
          kind: "VariableDefinition",
          variable: {
            kind: "Variable",
            name: { kind: "Name", value: "showCurrent" },
          },
          type: {
            kind: "NonNullType",
            type: {
              kind: "NamedType",
              name: { kind: "Name", value: "Boolean" },
            },
          },
        },
        {
          kind: "VariableDefinition",
          variable: {
            kind: "Variable",
            name: { kind: "Name", value: "showHourly" },
          },
          type: {
            kind: "NonNullType",
            type: {
              kind: "NamedType",
              name: { kind: "Name", value: "Boolean" },
            },
          },
        },
      ],
      selectionSet: {
        kind: "SelectionSet",
        selections: [
          {
            kind: "Field",
            name: { kind: "Name", value: "getWeatherData" },
            arguments: [
              {
                kind: "Argument",
                name: { kind: "Name", value: "latitude" },
                value: {
                  kind: "Variable",
                  name: { kind: "Name", value: "latitude" },
                },
              },
              {
                kind: "Argument",
                name: { kind: "Name", value: "longitude" },
                value: {
                  kind: "Variable",
                  name: { kind: "Name", value: "longitude" },
                },
              },
            ],
            selectionSet: {
              kind: "SelectionSet",
              selections: [
                { kind: "Field", name: { kind: "Name", value: "latitude" } },
                { kind: "Field", name: { kind: "Name", value: "longitude" } },
                {
                  kind: "FragmentSpread",
                  name: { kind: "Name", value: "WeatherCurrentF" },
                  directives: [
                    {
                      kind: "Directive",
                      name: { kind: "Name", value: "include" },
                      arguments: [
                        {
                          kind: "Argument",
                          name: { kind: "Name", value: "if" },
                          value: {
                            kind: "Variable",
                            name: { kind: "Name", value: "showCurrent" },
                          },
                        },
                      ],
                    },
                  ],
                },
                {
                  kind: "FragmentSpread",
                  name: { kind: "Name", value: "WeatherHourlyF" },
                  directives: [
                    {
                      kind: "Directive",
                      name: { kind: "Name", value: "include" },
                      arguments: [
                        {
                          kind: "Argument",
                          name: { kind: "Name", value: "if" },
                          value: {
                            kind: "Variable",
                            name: { kind: "Name", value: "showHourly" },
                          },
                        },
                      ],
                    },
                  ],
                },
              ],
            },
          },
        ],
      },
    },
    {
      kind: "FragmentDefinition",
      name: { kind: "Name", value: "WeatherCurrentF" },
      typeCondition: {
        kind: "NamedType",
        name: { kind: "Name", value: "WeatherData" },
      },
      selectionSet: {
        kind: "SelectionSet",
        selections: [
          {
            kind: "Field",
            name: { kind: "Name", value: "current" },
            selectionSet: {
              kind: "SelectionSet",
              selections: [
                { kind: "Field", name: { kind: "Name", value: "time" } },
                { kind: "Field", name: { kind: "Name", value: "interval" } },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "temperature2m" },
                },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "windSpeed10m" },
                },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "relativeHumidity2m" },
                },
              ],
            },
          },
          {
            kind: "Field",
            name: { kind: "Name", value: "currentUnits" },
            selectionSet: {
              kind: "SelectionSet",
              selections: [
                { kind: "Field", name: { kind: "Name", value: "time" } },
                { kind: "Field", name: { kind: "Name", value: "interval" } },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "temperature2m" },
                },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "windSpeed10m" },
                },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "relativeHumidity2m" },
                },
              ],
            },
          },
        ],
      },
    },
    {
      kind: "FragmentDefinition",
      name: { kind: "Name", value: "WeatherHourlyF" },
      typeCondition: {
        kind: "NamedType",
        name: { kind: "Name", value: "WeatherData" },
      },
      selectionSet: {
        kind: "SelectionSet",
        selections: [
          {
            kind: "Field",
            name: { kind: "Name", value: "hourly" },
            selectionSet: {
              kind: "SelectionSet",
              selections: [
                { kind: "Field", name: { kind: "Name", value: "time" } },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "temperature2m" },
                },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "relativeHumidity2m" },
                },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "windSpeed10m" },
                },
              ],
            },
          },
          {
            kind: "Field",
            name: { kind: "Name", value: "hourlyUnits" },
            selectionSet: {
              kind: "SelectionSet",
              selections: [
                { kind: "Field", name: { kind: "Name", value: "time" } },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "temperature2m" },
                },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "relativeHumidity2m" },
                },
                {
                  kind: "Field",
                  name: { kind: "Name", value: "windSpeed10m" },
                },
              ],
            },
          },
        ],
      },
    },
  ],
} as unknown as DocumentNode<GetWeatherDataQuery, GetWeatherDataQueryVariables>;
