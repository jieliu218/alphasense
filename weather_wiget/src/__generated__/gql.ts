import * as types from "./graphql";
import { TypedDocumentNode as DocumentNode } from "@graphql-typed-document-node/core";

/**
 * Map of all GraphQL operations in the project.
 *
 * This map has several performance disadvantages:
 * 1. It is not tree-shakeable, so it will include all operations in the project.
 * 2. It is not minifiable, so the string of a GraphQL query will be multiple times inside the bundle.
 * 3. It does not support dead code elimination, so it will add unused operations.
 *
 * Therefore it is highly recommended to use the babel or swc plugin for production.
 */
const documents = {
  "\n  fragment WeatherCurrentF on WeatherData {\n    current {\n      time\n      interval\n      temperature2m\n      windSpeed10m\n      relativeHumidity2m\n    }\n    currentUnits {\n      time\n      interval\n      temperature2m\n      windSpeed10m\n      relativeHumidity2m\n    }\n  }\n":
    types.WeatherCurrentFFragmentDoc,
  "\n  fragment WeatherHourlyF on WeatherData {\n    hourly {\n      time\n      temperature2m\n      relativeHumidity2m\n      windSpeed10m\n    }\n    hourlyUnits {\n      time\n      temperature2m\n      relativeHumidity2m\n      windSpeed10m\n    }\n  }\n":
    types.WeatherHourlyFFragmentDoc,
  "\n    query GetWeatherData(\n      $latitude: Float!\n      $longitude: Float!\n      $showCurrent: Boolean!\n      $showHourly: Boolean!\n    ) {\n      getWeatherData(latitude: $latitude, longitude: $longitude) {\n        latitude\n        longitude\n        ...WeatherCurrentF @include(if: $showCurrent)\n        ...WeatherHourlyF @include(if: $showHourly)\n      }\n    }\n  ":
    types.GetWeatherDataDocument,
};

/**
 * The gql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 *
 *
 * @example
 * ```ts
 * const query = gql(`query GetUser($id: ID!) { user(id: $id) { name } }`);
 * ```
 *
 * The query argument is unknown!
 * Please regenerate the types.
 */
export function gql(source: string): unknown;

/**
 * The gql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 */
export function gql(
  source: "\n  fragment WeatherCurrentF on WeatherData {\n    current {\n      time\n      interval\n      temperature2m\n      windSpeed10m\n      relativeHumidity2m\n    }\n    currentUnits {\n      time\n      interval\n      temperature2m\n      windSpeed10m\n      relativeHumidity2m\n    }\n  }\n",
): (typeof documents)["\n  fragment WeatherCurrentF on WeatherData {\n    current {\n      time\n      interval\n      temperature2m\n      windSpeed10m\n      relativeHumidity2m\n    }\n    currentUnits {\n      time\n      interval\n      temperature2m\n      windSpeed10m\n      relativeHumidity2m\n    }\n  }\n"];
/**
 * The gql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 */
export function gql(
  source: "\n  fragment WeatherHourlyF on WeatherData {\n    hourly {\n      time\n      temperature2m\n      relativeHumidity2m\n      windSpeed10m\n    }\n    hourlyUnits {\n      time\n      temperature2m\n      relativeHumidity2m\n      windSpeed10m\n    }\n  }\n",
): (typeof documents)["\n  fragment WeatherHourlyF on WeatherData {\n    hourly {\n      time\n      temperature2m\n      relativeHumidity2m\n      windSpeed10m\n    }\n    hourlyUnits {\n      time\n      temperature2m\n      relativeHumidity2m\n      windSpeed10m\n    }\n  }\n"];
/**
 * The gql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 */
export function gql(
  source: "\n    query GetWeatherData(\n      $latitude: Float!\n      $longitude: Float!\n      $showCurrent: Boolean!\n      $showHourly: Boolean!\n    ) {\n      getWeatherData(latitude: $latitude, longitude: $longitude) {\n        latitude\n        longitude\n        ...WeatherCurrentF @include(if: $showCurrent)\n        ...WeatherHourlyF @include(if: $showHourly)\n      }\n    }\n  ",
): (typeof documents)["\n    query GetWeatherData(\n      $latitude: Float!\n      $longitude: Float!\n      $showCurrent: Boolean!\n      $showHourly: Boolean!\n    ) {\n      getWeatherData(latitude: $latitude, longitude: $longitude) {\n        latitude\n        longitude\n        ...WeatherCurrentF @include(if: $showCurrent)\n        ...WeatherHourlyF @include(if: $showHourly)\n      }\n    }\n  "];

export function gql(source: string) {
  return (documents as any)[source] ?? {};
}

export type DocumentType<TDocumentNode extends DocumentNode<any, any>> =
  TDocumentNode extends DocumentNode<infer TType, any> ? TType : never;
