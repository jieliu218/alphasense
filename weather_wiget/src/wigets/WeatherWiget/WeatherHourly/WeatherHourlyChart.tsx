import React, { FC } from "react";
import { ApexOptions } from "apexcharts";
import {
  Box,
  useTheme,
  BoxProps,
  Divider,
  Stack,
  Typography,
} from "@mui/material";
import Chart, { useChart } from "../../../components/Chart";
import { bgBlur, textBlur } from "../../../theme/css";
import { fDateTime } from "../../../utils/formatTime";
import { fNumber } from "../../../utils/formatNumber";
import { Maybe } from "../../../__generated__/graphql";

interface WeatherHourlyChartProps extends BoxProps {
  title: string;
  icon: React.ReactNode;
  time: string[] | undefined;
  unit: Maybe<string> | undefined;
  chart: {
    colors?: string[];
    series: number[];
    options?: ApexOptions;
  };
}

const WeatherHourlyChart: FC<WeatherHourlyChartProps> = ({
  title,
  icon,
  time,
  unit,
  chart,
  sx,
  ...other
}) => {
  const theme = useTheme();

  const {
    colors = [theme.palette.primary.light, theme.palette.primary.main],
    series,
    options,
  } = chart;

  const chartOptions = useChart({
    colors: [colors[1]],
    fill: {
      type: "gradient",
      gradient: {
        colorStops: [
          // @ts-expect-error: ApexCharts type error
          { offsets: 0, color: colors[0] },
          { offsets: 100, color: colors[1] },
        ],
      },
    },
    chart: {
      animations: {
        enabled: true,
      },
      sparkline: {
        enabled: true,
      },
    },
    xaxis: {
      categories: (time || []).map((t) => fDateTime(t, "dd MMM p")),
    },
    tooltip: {
      x: {
        show: true,
      },
      y: {
        formatter: (value: number) => fNumber(value) + unit,
        title: {
          formatter: () => "",
        },
      },
      marker: {
        show: false,
      },
    },
    ...options,
  });

  const textColor = textBlur(theme.palette.common.white, 0.7);

  return (
    <Box
      sx={{
        display: "flex",
        flexDirection: "column",
        bgcolor: bgBlur({ color: theme.palette.common.white, opacity: 0.2 }),
        p: 2,
        borderRadius: 2,
        ...sx,
      }}
      {...other}
    >
      <Stack
        direction="row"
        justifyContent="flex-start"
        alignItems="center"
        spacing={0.25}
      >
        {icon}
        <Typography
          gutterBottom
          variant="body1"
          component="span"
          color={textColor}
        >
          {title}
        </Typography>
      </Stack>
      <Divider sx={{ mt: 2, mb: 2 }} />
      <Chart
        type="line"
        series={[{ data: series as number[] }]}
        options={chartOptions}
        width={"100%"}
        height={64}
      />
    </Box>
  );
};

export default WeatherHourlyChart;
