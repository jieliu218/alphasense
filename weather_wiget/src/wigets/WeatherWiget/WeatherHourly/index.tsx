import ThermostatIcon from "@mui/icons-material/Thermostat";
import AirIcon from "@mui/icons-material/Air";
import OpacityIcon from "@mui/icons-material/Opacity";
import { Box, Divider, Skeleton, Stack, useTheme } from "@mui/material";
import { gql, FragmentType, useFragment } from "../../../__generated__";
import { HourlyUnits, HourlyWeather } from "../../../__generated__/graphql";
import { textBlur } from "../../../theme/css";
import WeatherHourlyChart from "./WeatherHourlyChart";

export const WeatherHourlyFragment = gql(/* GraphQL */ `
  fragment WeatherHourlyF on WeatherData {
    hourly {
      time
      temperature2m
      relativeHumidity2m
      windSpeed10m
    }
    hourlyUnits {
      time
      temperature2m
      relativeHumidity2m
      windSpeed10m
    }
  }
`);

const WeatherHourly = (props: {
  hourly: FragmentType<typeof WeatherHourlyFragment>;
  hourlyUnits: FragmentType<typeof WeatherHourlyFragment>;
}) => {
  const theme = useTheme();

  const hourly = useFragment(
    WeatherHourlyFragment,
    props.hourly,
  ) as HourlyWeather;
  const hourlyUnits = useFragment(
    WeatherHourlyFragment,
    props.hourlyUnits,
  ) as HourlyUnits;

  const textColor = textBlur(theme.palette.common.white, 0.7);

  return (
    <Box data-testid="WeatherHourly">
      <Divider sx={{ mt: 2, mb: 2, color: textColor }}>Hourly</Divider>
      <Stack
        direction="column"
        justifyContent="justify-content"
        alignItems="center"
        spacing={1}
      >
        <WeatherHourlyChart
          title="Temperature"
          icon={<ThermostatIcon sx={{ fontSize: 20, color: textColor }} />}
          time={hourly?.time as string[]}
          unit={hourlyUnits?.temperature2m}
          chart={{
            series: hourly?.temperature2m as number[],
          }}
        />
        <WeatherHourlyChart
          title="Humidity"
          icon={<OpacityIcon sx={{ fontSize: 20, color: textColor }} />}
          time={hourly?.time as string[]}
          unit={hourlyUnits?.relativeHumidity2m}
          chart={{
            colors: [theme.palette.info.light, theme.palette.info.main],
            series: hourly?.relativeHumidity2m as number[],
          }}
        />
        <WeatherHourlyChart
          title="Wind Speed"
          icon={<AirIcon sx={{ fontSize: 20, color: textColor }} />}
          time={hourly?.time as string[]}
          unit={hourlyUnits?.windSpeed10m}
          chart={{
            colors: [
              theme.palette.secondary.light,
              theme.palette.secondary.main,
            ],
            series: hourly?.windSpeed10m as number[],
          }}
        />
      </Stack>
    </Box>
  );
};

const WeatherHourlySkeleton = ({ textColor }: { textColor: string }) => {
  return (
    <Box>
      <Divider sx={{ mt: 2, mb: 2, color: textColor }}>Hourly</Divider>
      <Stack
        direction="column"
        justifyContent="justify-content"
        alignItems="center"
        spacing={1}
      >
        <Skeleton variant="rectangular" width={332} height={145} />
        <Skeleton variant="rectangular" width={332} height={145} />
        <Skeleton variant="rectangular" width={332} height={145} />
      </Stack>
    </Box>
  );
};

export { WeatherHourlySkeleton };
export default WeatherHourly;
