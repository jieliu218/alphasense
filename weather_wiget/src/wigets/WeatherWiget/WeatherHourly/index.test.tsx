import { render, screen } from "@testing-library/react";
import WeatherHourly, { WeatherHourlyFragment } from "./index";
import { FragmentType, useFragment } from "../../../__generated__";

jest.mock("@mui/material", () => ({
  ...jest.requireActual("@mui/material"),
  useTheme: () => ({
    palette: {
      common: { white: "#ffffff" },
      primary: { light: "#f6f6f6", main: "#ffffff" },
      info: { light: "#abc", main: "#def" },
      secondary: { light: "#123", main: "#456" },
    },
  }),
}));

jest.mock("../../../__generated__", () => ({
  useFragment: jest.fn(),
  gql: jest.fn(),
}));

jest.mock("./WeatherHourlyChart", () => ({
  __esModule: true,
  default: jest
    .fn()
    .mockImplementation(({ title }) => <div>{title} Chart Mock</div>),
}));

const mockHourlyData = {
  time: ["2022-01-01T00:00:00Z", "2022-01-01T01:00:00Z"],
  temperature2m: [10, 11],
  relativeHumidity2m: [60, 65],
  windSpeed10m: [5, 10],
} as FragmentType<typeof WeatherHourlyFragment>;

const mockHourlyUnits = {
  temperature2m: "Â°C",
  relativeHumidity2m: "%",
  windSpeed10m: "km/h",
} as FragmentType<typeof WeatherHourlyFragment>;

describe("WeatherHourly", () => {
  it("renders correctly with hourly weather data", () => {
    (useFragment as jest.Mock)
      .mockReturnValueOnce(mockHourlyData)
      .mockReturnValueOnce(mockHourlyUnits);

    const { asFragment } = render(
      <WeatherHourly hourly={mockHourlyData} hourlyUnits={mockHourlyUnits} />,
    );

    expect(screen.getByText("Temperature Chart Mock")).toBeInTheDocument();
    expect(screen.getByText("Humidity Chart Mock")).toBeInTheDocument();
    expect(screen.getByText("Wind Speed Chart Mock")).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });
});
