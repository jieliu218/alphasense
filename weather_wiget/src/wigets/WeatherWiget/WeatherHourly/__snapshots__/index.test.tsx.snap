// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherHourly renders correctly with hourly weather data 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiDivider-root MuiDivider-fullWidth MuiDivider-withChildren css-1p87euv-MuiDivider-root"
      role="separator"
    >
      <span
        class="MuiDivider-wrapper css-qywfm8-MuiDivider-wrapper"
      >
        Hourly
      </span>
    </div>
    <div
      class="MuiStack-root css-1fcys0l-MuiStack-root"
    >
      <div>
        Temperature Chart Mock
      </div>
      <div>
        Humidity Chart Mock
      </div>
      <div>
        Wind Speed Chart Mock
      </div>
    </div>
  </div>
</DocumentFragment>
`;
